cdc/sink/ddlsink/mysql/mysql_ddl_sink.go

```diff
@@ -306,23 +357,116 @@ func needFormatDDL(db *sql.DB, cfg *pmysql.Config) bool {
 	return false
 }
 
-// WriteCheckpointTs does nothing.
-func (m *DDLSink) WriteCheckpointTs(_ context.Context, _ uint64, _ []*model.TableInfo) error {
-	// Only for RowSink for now.
-	return nil
+func getDDLCreateTime(ctx context.Context, db *sql.DB) int64 {
+	var ddlCreateTime int64 = -1 // default when scan failed
+	row, err := db.QueryContext(ctx, "SELECT UNIX_TIMESTAMP()")
+	if err != nil {
+		log.Warn("selecting unix timestamp failed", zap.Error(err))
+	} else {
+		for row.Next() {
+			err = row.Scan(&ddlCreateTime)
+			if err != nil {
+				log.Warn("getting ddlCreateTime failed", zap.Error(err))
+			}
+		}
+		//nolint:sqlclosecheck
+		_ = row.Close()
+		_ = row.Err()
+	}
+	return ddlCreateTime
 }
 
-// Close closes the database connection.
-func (m *DDLSink) Close() {
-	if m.statistics != nil {
-		m.statistics.Close()
-	}
-	if m.db != nil {
-		if err := m.db.Close(); err != nil {
-			log.Warn("MySQL ddl sink close db wit error",
-				zap.String("namespace", m.id.Namespace),
-				zap.String("changefeed", m.id.ID),
-				zap.Error(err))
+// getDDLStatusFromTiDB retrieves the synchronizing status of DDL from TiDB
+func getDDLStatusFromTiDB(ctx context.Context, db *sql.DB, ddl string, createTime int64) (string, error) {
+	linesOfRows := 10
+	rowNum := linesOfRows
+	rowOffset := 0
+	queryMap := make(map[int]string)
+
+	for {
+		// every attempt try 10 history jobs
+		showJobs := fmt.Sprintf("ADMIN SHOW DDL JOBS %d", rowNum)
```

Comment:
please use information_schema.ddl_jobs to get these infos instead of ADMIN SHOW DDL JOBS.
ref // TODO: Use the flollowing SQL to check the ddl job status after tidb optimize
// the information_schema.ddl_jobs table. Ref: https://github.com/pingcap/tidb/issues/55725
//
// SELECT JOB_ID, JOB_TYPE, SCHEMA_STATE, SCHEMA_ID, TABLE_ID, STATE, QUERY
// FROM information_schema.ddl_jobs